flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -447,10 +459,12 @@ private static void validateFlinkTable(CatalogBaseTable table) {
     if (!schema.getWatermarkSpecs().isEmpty()) {
       throw new UnsupportedOperationException("Creating table with watermark specs is not supported yet.");
     }
+  }
 
-    if (schema.getPrimaryKey().isPresent()) {
-      throw new UnsupportedOperationException("Creating table with primary key is not supported yet.");
-    }
+  private static List<String> toEqualityColumns(TableSchema schema) {
+    List<String> equalityColumns = Lists.newArrayList();
+    schema.getPrimaryKey().ifPresent(uniqueConstraint -> equalityColumns.addAll(uniqueConstraint.getColumns()));
```

Comment:
It seems like we should track primary key columns in the table format rather than in properties. If we are going to add primary key columns, then we should add it before v2 so that v2 writers are required to not drop it.